example/sound.cpp

```diff
@@ -4,25 +4,98 @@
 #include <dr_libs/dr_wav.h>
 
 #include <iostream>
+#include <type_traits>
 #include <vector>
 
+namespace {
+static constexpr int fail_code = 2;
+
+[[noreturn]] void on_error(ALenum error, int line) {
+	std::cerr << "Misc OpenAL error: " << std::hex << error << " on line " << std::dec << line << std::endl;
+	std::exit(fail_code);
+}
+
+void al_check(int line = __builtin_LINE()) {
+	if (auto err = alGetError(); err != AL_NO_ERROR) { on_error(err, line); }
+}
+
+class al_instance final {
```

Comment:
if we're sticking with PascalCase this should be `AlInstance` (or `ALInstance`)

Reply:
This PR doesn't cover design decisions.